src/Services/QueuedJobService.php

```diff
@@ -708,7 +708,11 @@ public function runJob($jobId)
                     }
 
                     if (!$broken) {
-                        // Collect output as job messages as well as sending it to the screen
+                        // Inject real-time logger
+                        $queuedJobLogger = new QueuedJobLogger($job, $jobDescriptor);
+                        Injector::inst()->registerService($queuedJobLogger, LoggerInterface::class);
```

Comment:
I'm not convinced that this is a good decision - you're replacing the LoggerInterface that exists already with your own implementation. What you should do is push new handlers into the stack of the existing LoggerInterface.

Reply:
Yeah, I thought about this as well. Don't like replacement of the global logger, but I could not come up with how to solve this.
I guess since Monolog is a framework requirement we could switch here to `Monolog\Logger` and simply add another handler then?

Reply:
Is the objective to use the core error logger? If so, just access it via its service name. Note the PR I mentioned earlier has been merged now, so the logger service name will be `LoggerInterface.core` or something from 4.4 onwards, which will make my team's life a little more complicated for things like this, but anyway...

You could also define your own logger - see https://docs.silverstripe.org/en/4/developer_guides/debugging/error_handling/

I'd suggest using the core error logger for now. `Injector::inst()->get(LoggerInterface::class)->info('hi')` etc

Reply:
Yeah, agree, it makes more sense for this service to use the core logger rather than registering a new one.
@bergice I guess that's been done in the first place just because 4.3 does not provide a default logger?